datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -3256,6 +3460,11 @@ public String toString() {
     public List<BrooklinMetricInfo> getMetricInfos() {
       return null;
     }
+
+    @Override
+    public List<String> getActiveTasks() {
```

Comment:
Have you added a test that covers the following scenarios:
a) Task list has some tasks for the datastream that eventually get cleared.
b) Task list has some tasks for the datastream that the task thread continues to run.

Reply:
(a) is covered in `testClaimAssignmentTokensForStoppingStreams`. Here I use the TestHookConnector to simulate that tasks have stopped (active tasks list is empty).
(b) is covered in `testTokensNotClaimedForConnectorThatFailedToStop`. Here the mock for Connector always keeps some of the tasks "running"